### ISSUE TITLE
Scripting languages

### ISSUE THREAD

**Issue by edfletcher** [nhexirc/client#8]
* Scripting languages (we mentioned bash-like/python/JS; I'd personally be also 100% content with just JS, but I think supporting different ones is attractive)
* If not already covered by scripts, nice ways to get around the icky parts of IRC (sending an invite without needing to op up, having that done for you automatically)

_Originally posted by @terrablue in https://github.com/nhexirc/client/discussions/1#discussioncomment-8908792_

**Comment by edfletcher** [comment]
> we mentioned bash-like/python/JS; I'd personally be also 100% content with just JS, but I think supporting different ones is attractive

Yep I'm of exactly the same mind: I want JS to be the first implemented and if that's all we end up getting to, then so be it. But ideally I'd love to have Python and Bash-like if possible... _really_ ideally would be able to support pre-existing HexChat scripts _without modification_ but that's a big stretch (and may dictate design decisions we're not happy with, so may not be a good idea after all)

**Comment by PantelicGR** [comment]
I would also like Nhex to be scriptable and to have Nhex Python Interface and Nhex Javascript Interface too.
Just as HexChat does. That's why it really is powerful and it customizes the behavior of the app.
I like Python very much and I have used it successfully in the past.
